Automattic/wp-calypso
# A4A Dev Sites: Add notice when missing payment data for referral dev sites

Fixes https://github.com/Automattic/dotcom-forge/issues/9134

## Proposed Changes

* Add notice when missing payment data for referral dev sites

![Markup on 2024-09-13 at 16:59:36](https://github.com/user-attachments/assets/a5abe9c4-95be-4124-a2a9-cd4a9801f29c)

## Why are these changes being made?

pdDOJh-3Cl-p2

## Testing Instructions

1. Apply the PR and build the app with `yarn start-a8c-for-agencies`.
2. Make sure you are proxied and with the Tipalti referral payment settings in the A4A dashboard set up (http://agencies.localhost:3000/referrals/dashboard).
3. Disable proxy.
4. Navigate to http://agencies.localhost:3000/marketplace/checkout?referral_blog_id={dev_blog_id}.
5. You should see the missing payment settings notice with the form fields disabled (as can be seen in the screenshot above). The `Go to payment settings` button should work as expected.
6. Enable proxy and reload the page. The notice should be gone and you should be able to submit the form.
7. If you go to the Referrals dashboard (http://agencies.localhost:3000/referrals/dashboard) when not proxied, the notice there should render correctly as well:

![Markup on 2024-09-18 at 16:49:13](https://github.com/user-attachments/assets/d27fd1dc-9726-46a9-a498-d20c831be30f)

## Pre-merge Checklist
